src/QGCApplication.cc

```diff
@@ -483,15 +483,29 @@ void QGCApplication::criticalMessageBoxOnMainThread(const QString& title, const
 
 void QGCApplication::saveTempFlightDataLogOnMainThread(QString tempLogfile)
 {
-    QString saveFilename = QGCFileDialog::getSaveFileName(
-        MainWindow::instance(),
-        tr("Save Flight Data Log"),
-        qgcApp()->mavlinkLogFilesLocation(),
-        tr("Flight Data Log Files (*.mavlink)"),
-        "mavlink");
-    if (!saveFilename.isEmpty()) {
-        QFile::copy(tempLogfile, saveFilename);
-    }
+    bool saveError;
+    do{
+        saveError = false;
+        QString saveFilename = QGCFileDialog::getSaveFileName(
+            MainWindow::instance(),
+            tr("Save Flight Data Log"),
+            qgcApp()->mavlinkLogFilesLocation(),
+            tr("Flight Data Log Files (*.mavlink)"),
+            "mavlink");
+    
+        if (!saveFilename.isEmpty()) {
+            // if file exsits already, try to remove it first to overwrite it
+            if(QFile::exists(saveFilename) && !QFile::remove(saveFilename)){
+                // if the file cannot be removed, prompt user and ask new path
+                saveError = true;
+                QGCApplication::criticalMessageBoxOnMainThread("File Error","Could not overwrite existing file.\nPlease provide a different file name to save to.");
```

Comment:
You're already on the main thread so you just need QGCApplication::warning. Warning instead of critical. Critical is more for you're about to crash type messages.